src/internal/client.ts

```diff
@@ -2810,4 +2836,182 @@ export class TypedClient {
       return await this.abortMultipartUpload(destObjConfig.Bucket, destObjConfig.Object, uploadId)
     }
   }
+
+  async presignedUrl(
+    method: string,
+    bucketName: string,
+    objectName: string,
+    expires?: number | PreSignRequestParams | undefined,
```

Comment:
Hi @prakashsvmx, shouldn't the expiry parameter just be a number instead of also having the object defined by `PreSignRequestParams` here?

```suggestion
    expires?: number,
```

Reply:
this can be optional and other parameters can be in place like `reqParams`. 
so additional checks for parameter validations (already it was existing) have been added between lines 2860-2863